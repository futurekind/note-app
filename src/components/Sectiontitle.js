import React, {PropTypes} from 'react'
import styled from 'styled-components';

import Icon from './Icon';
import Divider from './Divider';

const View =  styled.h2`
    margin: 0;
`

const Body = styled.div`
    display: flex;
    align-items: center;
`

const Label = styled.div`
    padding: 20px 0;
    flex: 1;
`

const ViewIcon = styled.div`
    padding-left: 20px;
    padding-right: 20px;
    display: flex;
    align-items: center;
`

const Sectiontitle = ({
    iconId,
    children
}) => {
    return (
        <View>
            <Body>
                <Label>{ children }</Label>
                { iconId && <ViewIcon>
                    <Icon id={ iconId } />   
                </ViewIcon> }
            </Body>
            <Divider />
        </View>
    )
}

Sectiontitle.propTypes = {
    iconId: PropTypes.string,
}

export default Sectiontitle