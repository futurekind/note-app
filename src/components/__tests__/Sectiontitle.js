import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Sectiontitle from '../Sectiontitle';

describe('Sectiontitle', () => {
    let wrapper;

    beforeAll(() => {
        wrapper = shallow(
            <Sectiontitle />
        )
    })

    it('renders', () => {
        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders children', () => {
        wrapper.setProps({
            children: 'Some Title'
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

    it('renders icon', () => {
        wrapper.setProps({
            iconId: 'edit',
        })

        expect(toJSON(wrapper)).toMatchSnapshot()
    })

})