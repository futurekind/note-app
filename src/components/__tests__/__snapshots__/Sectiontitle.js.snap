exports[`Sectiontitle renders 1`] = `
<styled.h2>
  <styled.div>
    <styled.div />
  </styled.div>
  <Divider />
</styled.h2>
`;

exports[`Sectiontitle renders children 1`] = `
<styled.h2>
  <styled.div>
    <styled.div>
      Some Title
    </styled.div>
  </styled.div>
  <Divider />
</styled.h2>
`;

exports[`Sectiontitle renders icon 1`] = `
<styled.h2>
  <styled.div>
    <styled.div>
      Some Title
    </styled.div>
    <styled.div>
      <Icon
        id="edit"
        size={24} />
    </styled.div>
  </styled.div>
  <Divider />
</styled.h2>
`;
